FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +20,31 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let self = self else { return }
+			switch result {
+			case let .success((data, response)):
+				let successResult = self.handleSuccess(data: data, response: response)
+				completion(successResult)
+			default:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+
+	// MARK: - Helpers
+	
+	private let OK_STATUS_CODE = 200
+	
+	private func handleSuccess(data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
+		guard response.statusCode == OK_STATUS_CODE else {
+			return .failure(Error.invalidData)
+		}
+		do {
+			let items = try FeedLoaderMapper.map(data: data)
+			return .success(items)
+		} catch {
+			return .failure(Error.invalidData)
+		}
```

Comment:
There's no need for explicit `do ... catch` block because the error that can emerge is not used. 
What this block does indeed is it maps the error that potentially can emerge on `try FeedLoaderMapper.map(data: data)` to `Error.invalidData`. It can be formulated in a more concise way using `try?`

Taking this into account and the `guard ... else` block at the top, the whole function can be shortened to the following:
```suggestion
		guard response.statusCode == OK_STATUS_CODE, let items = try? FeedLoaderMapper.map(data: data) else {
			return .failure(Error.invalidData)
		}
		return .success(items)
```